pkgs/applications/science/logic/klee/klee-uclibc.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, fetchurl
+, fetchFromGitHub
+, which
+, linuxHeaders
+, kleeClang
+, kleeLLVM
+, kleePython
+, debugRuntime ? true
+, runtimeAsserts ? false
+, extraKleeuClibcConfig ? {}
+}:
+
+let
+  localeSrcBase = "uClibc-locale-030818.tgz";
+  resolvedExtraKleeuClibcConfig = lib.mapAttrsToList (name: value: "${name}=${value}") (extraKleeuClibcConfig // {
+    "UCLIBC_DOWNLOAD_PREGENERATED_LOCALE_DATA" = "n";
+    "RUNTIME_PREFIX" = "/";
+    "DEVEL_PREFIX" = "/";
+  });
+in
+kleeClang.stdenv.mkDerivation rec {
+  pname = "klee-uclibc";
+  version = "1.2";
+  src = fetchFromGitHub {
+    owner = "klee";
+    repo = "klee-uclibc";
+    rev = "klee_uclibc_v${version}";
+    sha256 = "qdrGMw+2XwpDsfxdv6swnoaoACcF5a/RWgUxUYbtPrI=";
+  };
+  localeSrc = fetchurl {
+    url = "http://www.uclibc.org/downloads/${localeSrcBase}";
+    sha256 = "xDYr4xijjxjZjcz0YtItlbq5LwVUi7k/ZSmP6a+uvVc=";
+  };
+  nativeBuildInputs = [
+    kleeClang
+    kleeLLVM
+    kleePython
+    which
+  ];
+
+  # No need - this is a bitcode build.
+  hardeningDisable = [ "all" ];
```

Comment:
Not a fan of this. Maybe we can get away with only disabling some hardening flags?

Reply:
klee-uclibc is a bitcode only build. I don't think hardening is really necessary since it's symbolically executed and would just slow things down.

Reply:
I am not that into compilters but I just trust you here.

Reply:
I removed it. I thought it was also causing problems I was running into but turns out it wasn't.